x/sequencer/types/metadata.go

```diff
@@ -0,0 +1,90 @@
+package types
+
+import errorsmod "cosmossdk.io/errors"
+
+// constant for maximum string length of the SequencerMetadata fields
+const (
+	MaxMonikerLength      = 70
+	MaxContactFieldLength = 140
+	MaxDetailsLength      = 280
+	MaxExtraDataLength    = 280
+)
+
+// DoNotModifyDesc constant is used in flags to indicate that the metadata field should not be updated
+const DoNotModifyDesc = "[do-not-modify]"
+
+// UpdateSequencerMetadata updates the fields of a given metadata. An error is
+// returned if the resulting metadata contains an invalid length.
+func (d SequencerMetadata) UpdateSequencerMetadata(d2 SequencerMetadata) (SequencerMetadata, error) {
+	metadata := SequencerMetadata{
+		Moniker:        d2.Moniker,
+		Details:        d2.Details,
+		P2PSeeds:       d2.P2PSeeds,
+		Rpcs:           d2.Rpcs,
+		EvmRpcs:        d2.EvmRpcs,
+		RestApiUrl:     d2.RestApiUrl,
+		ExplorerUrl:    d2.ExplorerUrl,
+		GenesisUrls:    d2.GenesisUrls,
+		ContactDetails: d2.ContactDetails,
+		ExtraData:      d2.ExtraData,
+		Snapshots:      d2.Snapshots,
+		GasPrice:       d2.GasPrice,
+	}
```

Comment:
so you cant update p2p or rpc or evm or rest api or explorer or genesis or snapshots or gas price?

Reply:
@zale144 ?